File: src/syntheticExamples/lazyInitialization/JDGUI/LazyInitializationDecompiled/LazyInitialization.java
package syntheticExamples.lazyInitialization.JDGUI.LazyInitializationDecompiled;

//'kotlin. Function' cannot be inherited as a raw type and with generic type arguments 'java. lang. String'

/*
@Metadata(mv = {2, 0, 0}, k = 1, xi = 82, d1 = {"\000\024\n\002\030\002\n\002\020\000\n\002\b\003\n\002\020\016\n\002\b\005\030\0002\0020\001B\007\006\004\b\002\020\003R\033\020\004\032\0020\0058FX\002\006\f\n\004\b\b\020\t\032\004\b\006\020\007\006\n"}, d2 = {"LsyntheticExamples/lazyInitialization/LazyInitialization;", "", "<init>", "()V", "value", "", "getValue", "()Ljava/lang/String;", "value$delegate", "Lkotlin/Lazy;", "KotlinDecompiler"})
public final class LazyInitialization {
    @NotNull
    private final Lazy value$delegate = LazyKt.lazy(LazyInitialization$value$2.INSTANCE);

    @NotNull
    public final String getValue() {
        Lazy lazy = this.value$delegate;
        return (String)lazy.getValue();
    }

    @Metadata(mv = {2, 0, 0}, k = 3, xi = 82, d1 = {"\000\b\n\000\n\002\020\016\n\000\020\000\032\0020\001H\n\006\002\b\002"}, d2 = {"<anonymous>", "", "invoke"})
    static final class LazyInitialization$value$2 extends Lambda implements Function0<String> {
        public static final LazyInitialization$value$2 INSTANCE = new LazyInitialization$value$2();

        @NotNull
        public final String invoke() {
            System.out.println("Initializing");
            return "Initialized";
        }

        LazyInitialization$value$2() {
            super(0);
        }
    }
}
 */

File: src/syntheticExamples/lazyInitialization/CFR/LazyInitializationJ2K/LazyInitialization.kt
package syntheticExamples.lazyInitialization.CFR.LazyInitializationJ2K

/*
@Metadata(
    mv = [2, 0, 0],
    k = 1,
    xi = 82,
    d1 = ["\u0000\u0014\n\u0002\u0018\u0002\n\u0002\u0010\u0000\n\u0002\b\u0003\n\u0002\u0010\u000e\n\u0002\b\u0005\u0018\u00002\u00020\u0001B\u0007\u00a2\u0006\u0004\b\u0002\u0010\u0003R\u001b\u0010\u0004\u001a\u00020\u00058FX\u0086\u0084\u0002\u00a2\u0006\u000c\n\u0004\b\b\u0010\t\u001a\u0004\b\u0006\u0010\u0007\u00a8\u0006\n"],
    d2 = ["LsyntheticExamples/lazyInitialization/LazyInitialization;", "", "<init>", "()V", "value", "", "getValue", "()Ljava/lang/String;", "value\$delegate", "Lkotlin/Lazy;", "KotlinDecompiler"]
)
 */

// not working
/*
class LazyInitialization {
    private val `value$delegate`: Lazy<*> = lazy<Any>(value as Function0<*>)
    val value: String
        get() {
            val lazy = this.`value$delegate`
            return lazy.value as String
        }
}
 */